system/doc/design_principles/sup_princ.xml

```diff
@@ -637,5 +655,69 @@ supervisor:terminate_child(Sup, Pid)</code>
       the same procedure as described above, that is, the supervisor
       terminates all remaining child processes in reversed start
       order, and then terminates itself.</p>
+    <section>
+      <title>Manual stopping versus Automatic Shutdown</title>
+      <p>For several reasons, a supervisor should not be stopped
+        manually via of <c>supervisor:terminate_child/2</c> from
+        a child located in it's own tree.</p>
+      <list type="ordered">
+        <item>
+          <p>The child process will have to know the pids or registered
+            names not only of the supervisor it wants to stop, but also
+            that of the supervisor's parent supervisor, in order to tell
+            it to stop the supervisor it wants to stop. This can make
```

Comment:
Bit confusing "it".

Reply:
Better now?

Reply:
Better but there's still "stop the supervisor it wants to stop". But I don't have a suggestion.

Reply:
I'll leave it at that for now.